app/controllers/reservations_controller.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+class ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show]
+
+  def index
+    reservations = Reservation.all
+
+    render json: reservations
+  end
+
+  def show
+    render json: @reservation
+  end
+
+  def create
```

Comment:
Consider transaction also here.

Reply:
I double-thought on both cases and added transaction.